src/commands/admin/prune.ts

```diff
@@ -0,0 +1,107 @@
+import { Command } from '@lib/types/Command';
+import { ROLES } from '@root/config';
+import { adminPerms } from '@root/src/lib/permissions';
+import { GuildMember, Message, MessageActionRow, MessageButton, MessageEmbed } from 'discord.js';
+
+const PRUNE_TIMEOUT = 30;
+
+export default class extends Command {
+
+	description = `Prunes all members who don't have the <@&${ROLES.VERIFIED}> role`;
+	runInDM = false;
+
+	permissions(msg: Message): boolean {
+		return adminPerms(msg);
+	}
+
+	async run(msg: Message): Promise<void> {
+		let timeout = PRUNE_TIMEOUT;
+		await msg.guild.members.fetch();
+		const toKick = msg.guild.members.cache.filter(member => !member.user.bot && !member.roles.cache.has(ROLES.VERIFIED));
+
+		const confirmEmbed = new MessageEmbed()
+			.setTitle(`Server prune will kick ${toKick.size} members from the guild. Proceed?`)
+			.setColor('RED')
+			.setFooter(`This command will expire in ${PRUNE_TIMEOUT} seconds`);
+
+		const confirmBtns = [
+			new MessageButton({ label: 'Cancel', customId: 'cancel', style: 'SECONDARY' }),
+			new MessageButton({ label: 'Proceed', customId: 'proceed', style: 'DANGER' })
+		];
+
+		const confirmMsg = await msg.channel.send({
+			embeds: [confirmEmbed],
+			components: [new MessageActionRow({ components: confirmBtns })]
+		});
+
+		const collector = msg.channel.createMessageComponentCollector({
+			filter: i => i.message.id === confirmMsg.id, time: PRUNE_TIMEOUT * 1000
+		});
+
+		const countdown = setInterval(() => this.countdown(confirmMsg, --timeout, confirmBtns, confirmEmbed), 1000);
+
+		collector.on('collect', async interaction => {
+			if (!interaction.isButton()) return;
+			if (interaction.user.id !== msg.author.id) {
+				await interaction.reply({
+					content: 'You cannot respond to a command you did not execute',
```

Comment:
does this stop the countdown completely? I assume it does not from the tests you performed in the testing server, but the return just below makes me somewhat think this.

This is more a question of curiosity rather than me pointing out a problem here.

Reply:
It does not, the return stops execution for that specific interaction but not the interaction listening.

Reply:
That was my second thought. Appreciate the clarification!